From 828055a7dbf3653fa97bde566be0e5a556080067 Mon Sep 17 00:00:00 2001
From: Jerry Jelinek <jerry.jelinek@joyent.com>
Date: Mon, 10 Apr 2017 15:26:30 +0000
Subject: [PATCH] OS-6034 cgroups and sysfs have wrong statfs f_type magic
 number Reviewed by: Patrick Mooney <patrick.mooney@joyent.com> Approved by:
 Patrick Mooney <patrick.mooney@joyent.com>

---
 usr/src/lib/brand/lx/lx_brand/common/statfs.c | 4 +++-
 usr/src/lib/brand/lx/lx_brand/sys/lx_statfs.h | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/usr/src/lib/brand/lx/lx_brand/common/statfs.c b/usr/src/lib/brand/lx/lx_brand/common/statfs.c
index 942aaf9882..629e549831 100644
--- a/usr/src/lib/brand/lx/lx_brand/common/statfs.c
+++ b/usr/src/lib/brand/lx/lx_brand/common/statfs.c
@@ -21,7 +21,7 @@
 /*
  * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
- * Copyright 2015 Joyent, Inc.  All rights reserved.
+ * Copyright 2017 Joyent, Inc.  All rights reserved.
  */
 
 #include <assert.h>
@@ -105,6 +105,8 @@ static struct lx_ftype_name {
 	{ "lofs",	LX_EXT2_SUPER_MAGIC,	"LX_EXT2_SUPER_MAGIC"	},
 	{ "zfs",	LX_EXT2_SUPER_MAGIC,	"LX_EXT2_SUPER_MAGIC"	},
 	{ "lxautofs",	LX_AUTOFS_SUPER_MAGIC,	"LX_AUTOFS_SUPER_MAGIC"	},
+	{ "lx_cgroup",	LX_CGROUP_SUPER_MAGIC,	"LX_CGROUP_SUPER_MAGIC"	},
+	{ "lx_sysfs",	LX_SYSFS_SUPER_MAGIC,	"LX_SYSFS_SUPER_MAGIC"	},
 	{ NULL,		0,			NULL	}
 };
 
diff --git a/usr/src/lib/brand/lx/lx_brand/sys/lx_statfs.h b/usr/src/lib/brand/lx/lx_brand/sys/lx_statfs.h
index dd046398c4..02f1a9b32b 100644
--- a/usr/src/lib/brand/lx/lx_brand/sys/lx_statfs.h
+++ b/usr/src/lib/brand/lx/lx_brand/sys/lx_statfs.h
@@ -21,7 +21,7 @@
 /*
  * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
- * Copyright 2015 Joyent, Inc.  All rights reserved.
+ * Copyright 2017 Joyent, Inc.  All rights reserved.
  */
 
 #ifndef	_LX_STATFS_H
@@ -65,6 +65,7 @@ struct lx_statfs64 {
  * These magic values are taken mostly from statfs(2) or magic.h
  */
 #define	LX_AUTOFS_SUPER_MAGIC		0x0187
+#define	LX_CGROUP_SUPER_MAGIC		0x27e0eb
 #define	LX_DEVFS_SUPER_MAGIC		0x1373
 #define	LX_DEVPTS_SUPER_MAGIC		0x1cd1
 #define	LX_EXT2_SUPER_MAGIC		0xEF53
@@ -72,6 +73,7 @@ struct lx_statfs64 {
 #define	LX_MSDOS_SUPER_MAGIC		0x4d44
 #define	LX_NFS_SUPER_MAGIC		0x6969
 #define	LX_PROC_SUPER_MAGIC		0x9fa0
+#define	LX_SYSFS_SUPER_MAGIC		0x62656572
 #define	LX_TMPFS_SUPER_MAGIC		0x01021994
 #define	LX_UFS_MAGIC			0x00011954
 #define	LX_PIPEFS_MAGIC			0x50495045
-- 
2.21.0

